assignment-client/src/audio/AudioMixerClientData.cpp

```diff
@@ -88,21 +89,25 @@ int AudioMixerClientData::parseData(const QByteArray& packet) {
             // this is injected audio
 
             // grab the stream identifier for this injected audio
-            int bytesBeforeStreamIdentifier = numBytesForPacketHeader(packet) + sizeof(quint16);
-            QUuid streamIdentifier = QUuid::fromRfc4122(packet.mid(bytesBeforeStreamIdentifier, NUM_BYTES_RFC4122_UUID));
-            int bytesBeforeStereoIdentifier = bytesBeforeStreamIdentifier + NUM_BYTES_RFC4122_UUID;
+            packet.seek(sizeof(quint16));
+            QUuid streamIdentifier = QUuid::fromRfc4122(packet.read(NUM_BYTES_RFC4122_UUID));
+
             bool isStereo;
-            QDataStream(packet.mid(bytesBeforeStereoIdentifier)) >> isStereo;
+            packet.readPrimitive(&isStereo);
 
             if (!_audioStreams.contains(streamIdentifier)) {
                 // we don't have this injected stream yet, so add it
-                _audioStreams.insert(streamIdentifier, matchingStream = new InjectedAudioStream(streamIdentifier, isStereo, AudioMixer::getStreamSettings()));
+                _audioStreams.insert(streamIdentifier,
+                        matchingStream = new InjectedAudioStream(streamIdentifier, isStereo, AudioMixer::getStreamSettings()));
             } else {
                 matchingStream = _audioStreams.value(streamIdentifier);
             }
         }
 
-        return matchingStream->parseData(packet);
+        // seek to the beginning of the packet so that the next reader is in the right spot
```

Comment:
Doesn't really feel safe, a new reader should seek(0) before reading if it expects it to be at the beginning.